### DISCUSSION TITLE
How to get the field and method level dependency graph?

### DISCUSSION THREAD

**Discussion by rokii** [classgraph/classgraph#634] [General]
In Java IDE, we usually can get all the references/invocations to one field/method.

Can ClassGraph expose such information?
Or is there any idea on how to implement this?

**Comment by lukehutch** [comment]
Just to clarify, do you need to know which methods call which other methods, and which methods access which fields?

If so, ClassGraph cannot currently provide that information, because it only parses class metadata (so it can see which classes are referenced by which other classes). There are method and field names in the constant pool of a class, but those could be referenced by any method in the class, and to know which specific methods call or access which other methods or fields, you would have to parse the JVM bytecode itself. You would need ObjectWeb ASM to accomplish that. Parsing bytecode would add so much additional complexity to ClassGraph that it's really not worth duplicating this functionality in ClassGraph.

Sorry that I don't have better news on this!
